src/libsystemd/sd-journal/journal-send.c

```diff
@@ -61,10 +60,11 @@ static int journal_fd(void) {
 
         fd_inc_sndbuf(fd, SNDBUF_SIZE);
 
-        if (!__atomic_compare_exchange_n(&fd_plus_one, &(int){0}, fd+1,
-                false, __ATOMIC_SEQ_CST, __ATOMIC_SEQ_CST)) {
+        tmp_fd = 0;
+        if (!__atomic_compare_exchange_n(&fd_plus_one, &tmp_fd, fd+1,
+                false, __ATOMIC_SEQ_CST, __ATOMIC_RELAXED)) {
                 safe_close(fd);
-                goto retry;
+                fd = tmp_fd - 1;
```

Comment:
i think an `assert(tmp_fd > 0);` would make sense here first, to encode expectations here, and guard against the overflow